src/foam/u2/grid/GridCell_old.js

```diff
@@ -0,0 +1,347 @@
+            name: 'order', 
+        }, 
+    
+        'colProperty',
+        'rowProperty',
+        'colMatch',
+        'rowMatch',
+        //might not be an id, might be an object, date, etc. 
+        {
+            name: 'colMatchId',
+            expression: function(colMatch, matchColId){
+               if (matchColId && colMatch ){
+                return colMatch.id; 
+               }else {
+                return colMatch; 
+               }
+            }
+        }, 
+        {
+            name: 'rowMatchId',
+            expression: function(rowMatch, matchRowId){
+               if (matchRowId && rowMatch ){
+                return rowMatch.id; 
+               }else {
+                return rowMatch; 
+               }
+            }
+        }, 
+        {
+            name: 'selected',
+            value: false, 
+        },
+        {
+            name: 'matchRowId',
+            value: false, 
+        },
+        {
+            name: 'matchColId',
+            value: false, 
+        }, 
+        {
+            name: 'inSelectedRow',
+            value: false,
+        },
+        {
+            name: 'inSelectedCol',
+            value: false,
+        },
+        {
+            name: 'selectedCSSClass',
+            expression: function(selected){
+                return selected?this.myCls("selected"):"";
+            }
+        },
+        {
+            name: 'rowHighlightCSSClass',
+            expression: function(inSelectedRow, inSelectedCol){
+                if (inSelectedCol){
+                    return inSelectedRow?this.myCls("intersection-highlight"):"";
+                }else {
+                    return inSelectedRow?this.myCls("row-highlight"):"";
+                }
+            }
+        },
+        {
+            name: 'colHighlightCSSClass',
+            expression: function(inSelectedCol, inSelectedRow){
+                if (inSelectedRow){
+                    return inSelectedCol?this.myCls("intersection-highlight"):"";
+                }else {
+                    return inSelectedCol?this.myCls("col-highlight"):"";
+                }
+            }
+        },
+        
+        {
+            name: 'rowPredicate',
+            expression: function(rowProperty, rowMatchId, matchRowId){
+              return this.makePredicate(rowProperty, rowMatchId, matchRowId);
+            }
+        },
+        {
+            name: 'colPredicate',
+            expression: function(colProperty, colMatchId, matchColId){
+              return this.makePredicate(colProperty, colMatchId, matchColId);
+            }
+        },
+        {
+            name: 'predicate',
+            expression: function(rowPredicate, colPredicate){
+                var M = foam.mlang.Expressions.create();
+                return M.AND(rowPredicate, colPredicate);
+            }
+        },
+        
+        {
+            name: 'data',
+        },
+        
+        {
+            name: 'resultArr',
+            value: [], 
```

Comment:
This could be an issue if you create more than one instance of this object, they'll all share the same array for resultArr. Better to do factory: function() { return []; } to make sure they each get their own array, or, if possible, make the class: be of some Array type, so that will automatically be done for you.